v2/src/log_context_property_type_wchar_t_ptr.c

```diff
@@ -0,0 +1,151 @@
+        /* Codes_SRS_LOG_CONTEXT_PROPERTY_TYPE_WCHAR_T_PTR_07_009: [ LOG_CONTEXT_PROPERTY_TYPE_IF_IMPL(wchar_t_ptr).copy shall copy the entire string (including the null terminator) from src_value to dst_value. ]*/
+        (void)wcscpy(dst_value, src_value);
+        /* Codes_SRS_LOG_CONTEXT_PROPERTY_TYPE_WCHAR_T_PTR_07_010: [ LOG_CONTEXT_PROPERTY_TYPE_IF_IMPL(wchar_t_ptr).copy shall succeed and return 0. ]*/
+        result = 0;
+    }
+
+    return result;
+}
+
+static void wchar_t_ptr_log_context_property_type_free(void* value)
+{
+    /* Codes_SRS_LOG_CONTEXT_PROPERTY_TYPE_WCHAR_T_PTR_07_011: [ LOG_CONTEXT_PROPERTY_TYPE_IF_IMPL(wchar_t_ptr).free shall return. ]*/
+    (void)value;
+}
+
+static LOG_CONTEXT_PROPERTY_TYPE wchar_t_ptr_log_context_property_type_get_type(void)
+{
+    /* Codes_SRS_LOG_CONTEXT_PROPERTY_TYPE_WCHAR_T_PTR_07_012: [ LOG_CONTEXT_PROPERTY_TYPE_IF_IMPL(wchar_t_ptr).get_type shall return the property type LOG_CONTEXT_PROPERTY_TYPE_wchar_t_ptr. ]*/
+    return LOG_CONTEXT_PROPERTY_TYPE_wchar_t_ptr;
+}
+
+int LOG_CONTEXT_PROPERTY_TYPE_INIT(wchar_t_ptr)(void* dst_value, size_t count, const wchar_t* format, ...)
+{
+    int result;
+
+    if (
+        /* Codes_SRS_LOG_CONTEXT_PROPERTY_TYPE_WCHAR_T_PTR_07_013: [ If dst_value is NULL, LOG_CONTEXT_PROPERTY_TYPE_INIT(wchar_t_ptr) shall fail and return a non-zero value. ]*/
+        (dst_value == NULL) ||
+        /* Codes_SRS_LOG_CONTEXT_PROPERTY_TYPE_WCHAR_T_PTR_07_014: [ If format is NULL, LOG_CONTEXT_PROPERTY_TYPE_INIT(wchar_t_ptr) shall fail and return a non-zero value. ]*/
+        (format == NULL)
+        )
+    {
+        (void)printf("Invalid arguments: void* dst_value=%p, const wchar_t* format=%ls\r\n",
+            dst_value, MU_WP_OR_NULL(format));
+        result = MU_FAILURE;
+    }
+    else
+    {
+        va_list args;
+        va_start(args, format);
+
+        /* Codes_SRS_LOG_CONTEXT_PROPERTY_TYPE_WCHAR_T_PTR_07_015: [ LOG_CONTEXT_PROPERTY_TYPE_INIT(wchar_t_ptr) shall initialize by calling vswprintf the memory at dst_value with the wprintf style formatted wchar_t string given by format and the arguments in .... ]*/
+        int vswprintf_result = vswprintf(dst_value, count, format, args);
+
+        if (vswprintf_result < 0)
+        {
+        /* Codes_SRS_LOG_CONTEXT_PROPERTY_TYPE_WCHAR_T_PTR_07_017: [ If formatting the string fails, LOG_CONTEXT_PROPERTY_TYPE_INIT(wchar_t_ptr) shall fail and return a non-zero value. ]*/
+            result = MU_FAILURE;
+        }
+        else
+        {
+            /* Codes_SRS_LOG_CONTEXT_PROPERTY_TYPE_WCHAR_T_PTR_07_016: [ LOG_CONTEXT_PROPERTY_TYPE_INIT(wchar_t_ptr) shall succeed and return 0. ]*/
+            result = 0;
+        }
+
+        va_end(args);
+    }
+
+    return result;
+}
+
+int LOG_CONTEXT_PROPERTY_TYPE_GET_INIT_DATA_SIZE(wchar_t_ptr)(const wchar_t* format, ...)
+{
+    va_list args;
+    va_start(args, format);
+
+    wchar_t* buffer = (wchar_t*)malloc(sizeof(wchar_t)* LOG_MAX_WCHAR_STRING_LENGTH);
```

Comment:
>malloc [](http://example.com/codeflow?start=32&length=6)

Cannot be used, make it a static array here.

Reply:
"and everywhere" if there are other uses of malloc/other memory allocation functions

Reply:
modified to use static memory here, only here use malloc